### PR TITLE
Disable MDNS lookup on first message deliver failure be default

### PR DESCRIPTION
**Pull request by tehampson**
#### Issue Being Resolved
* Addresses #22670 in a manner that allows platforms/application to re-enable should they choose in 1.0 timeframe

#### Change overview
* Disable MDNS lookup on first message deliver failure be default, but allow it to be enabled via gn argument

### CODE REVIEW

File: src/messaging/BUILD.gn [hunk 1]
```diff
@@ -17,6 +17,8 @@ import("//build_overrides/chip.gni")
 declare_args() {
   # Allows to change time between retries during the case session
   chip_case_retry_delta = ""
+
+  chip_config_mrp_on_first_msg_fail_do_mdsn_lookup = ""
```
Comment (tcarmelveilleux) [inline src/messaging/BUILD.gn:21]: Nit: `chip_config_resolve_peer_on_first_transmit_failure`

**Comment by andy31415** [comment]
accepted for master: tied to an issue with 'master patches acceptable'

File: src/messaging/ReliableMessageProtocolConfig.h [hunk 2]
```diff
@@ -71,6 +71,22 @@ namespace chip {
...
+/**
+ *  @def CHIP_CONFIG_MRP_ON_FIRST_MSG_FAIL_DO_MDNS_LOOKUP
+ *
+ *  @brief
+ *    Should an address lookup of the peer happen on every first message that fails
+ *    to send on the link.
+ *
+ *  The default value selected to not perform lookup was for the following reasons:
+ *    1. The likelihood of the IP address changing right away is extremely slim.
+ *    2. On bad links, there resolutions occur extremely frequently, making the link
```
Comment (tcarmelveilleux) [inline src/messaging/ReliableMessageProtocolConfig.h:83]:
```suggestion
 *    2. On bad links, thsre resolutions occur extremely frequently, making the link
```

File: src/messaging/ReliableMessageProtocolConfig.h [hunk 3]
```diff
@@ -71,6 +71,22 @@ namespace chip {
...
+ *       worse.
```
Comment (tcarmelveilleux) [inline src/messaging/ReliableMessageProtocolConfig.h:84]:
```suggestion
 *       worse.
 *    3. Every message that arrives from a peer updates the address. If the peer has fallen off the link due to
 *       any other reason, a re-resolve may not achieve an address that is reachable, even if a resolve response
 *       occurs.
```

Comment (mrjerryjohns) [inline src/messaging/ReliableMessageProtocolConfig.h]:
```suggestion
 *    2. MRP re-transmissions usually happen when network conditions have degraded. Doing further DNS resolutions on a degraded link can exacerbate that problem greatly.
```

File: src/messaging/ReliableMessageProtocolConfig.h [hunk 4]
```diff
@@ -71,6 +71,22 @@ namespace chip {
...
 
+/**
+ *  @def CHIP_CONFIG_MRP_ON_FIRST_MSG_FAIL_DO_MDNS_LOOKUP
+ *
+ *  @brief
+ *    Should an address lookup of the peer happen on every first message that fails
+ *    to send on the link.
+ *
+ *  The default value selected to not perform lookup was for the following reasons:
+ *    1. The likelihood of the IP address changing right away is extremely slim.
```
Comment (mrjerryjohns) [inline src/messaging/ReliableMessageProtocolConfig.h]: Not sure what 'right away' is relative to.

```suggestion
 *    1. There is insufficient empirical data to indicate the frequency of occurrence of IP addresses changing *without* a reboot. Doing this to benefit a possibly small set of use-cases while causing harm in what is a fairly common scenario of encountering degraded networking conditions doesn't seem justified.
```

Comment (mrjerryjohns) [reply src/messaging/ReliableMessageProtocolConfig.h]: This could come after point 2) below.

**Review by tcarmelveilleux — APPROVED** [review]

**Review by mrjerryjohns — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**